### PR TITLE
Support for PermitAll and DenyAll annotation added.

### LINKED ISSUE #194: Security - add support for @RolesAllowed, @PermitAll, @DenyAll
Currently helidon MP supports RolesAllowed.
It would be great if PermitAll and DenyAll were also supported out of the box.

PermiAll - anybody has access (even users not logged in)
DenyAll - no-one has access

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #194

### CODE REVIEW

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public AnalyzerResponse analyze(Method maybeAnnotated, AnalyzerResponse previousResponse) {
+        return analyze(maybeAnnotated.getAnnotation(PermitAll.class), previousResponse);
+
+    }
+
+    private static AnalyzerResponse analyze(PermitAll permitAll, AnalyzerResponse previousResponse) {
+        if (permitAll == null) {
```
Comment (tomas-langer) [inline security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java]: if permitAll is null and previousResponse is configured (e.g. from class analysis), you just discard it.
The `previousResponse` must be preserved.

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            return AnalyzerResponse.abstain();
+        }
+
+        return AnalyzerResponse.builder(previousResponse)
+                .authenticationResponse(Flag.OPTIONAL)
+                .build();
```
Comment (tomas-langer) [inline security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java]: PermitAll disables authorization and marks authentication as optional - missing the authorization part here

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java [hunk 3]
```diff
@@ -84,37 +88,51 @@ public RoleConfig fromConfig(Config config) {
...
-                .build();
-    }
-
     @Override
     public void validate(RoleConfig config, Errors.Collector collector, ProviderRequest request) {
+        if (config.permitAll()) {
+            return;
+        }
+        if (config.denyAll()) {
+            collector.fatal(this, "This target is not accessible by any means!");
```
Comment (tomas-langer) [inline security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java]: This message is useless. Please use something like `Access denied by DenyAll`

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java [hunk 4]
```diff
@@ -329,6 +372,28 @@ private Builder addServiceRole(String role) {
...
+             * Sets if all access should be denied.
+             *
+             * @param denyAll if access should be denied for all
+             * @return updated builder instance
+             */
+            private Builder denyAll(boolean denyAll) {
+                this.denyAll = denyAll;
+                return this;
+            }
+
```
Comment (tomas-langer) [inline security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java]: Missing possibility to add permitAll and denyAll from configuration - method config(Config)

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java (hunk 2 above)
Comment (Verdent) [reply security/abac/role/src/main/java/io/helidon/security/abac/role/RoleAnnotationAnalyzer.java]: Authorization set to Optional added

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java (hunk 3 above)
Comment (Verdent) [reply security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java]: Message edited according to your suggestion.

File: security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java (hunk 4 above)
Comment (Verdent) [reply security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java]: Support for handling permitAll and denyAll from config implemented. (Tests included)

**Pull request merged**